jgrapht-core/src/main/java/org/jgrapht/alg/scoring/AlphaCentrality.java

```diff
@@ -0,0 +1,281 @@
+    }
+
+    /**
+     * Create and execute an instance of AlphaCentrality.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     */
+    public AlphaCentrality(Graph<V, E> g, double dampingFactor)
+    {
+        this(g, dampingFactor, EXOGENOUS_FACTOR_DEFAULT, MAX_ITERATIONS_DEFAULT, TOLERANCE_DEFAULT);
+    }
+
+    /**
+     * Create and execute an instance of AlphaCentrality.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     * @param exogenousFactor the exogenous factor
+     */
+    public AlphaCentrality(Graph<V, E> g, double dampingFactor, double exogenousFactor)
+    {
+        this(g, dampingFactor, exogenousFactor, MAX_ITERATIONS_DEFAULT, TOLERANCE_DEFAULT);
+    }
+
+    /**
+     * Create and execute an instance of AlphaCentrality.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     * @param exogenousFactormap the exogenous factor map
+     */
+    public AlphaCentrality(Graph<V, E> g, double dampingFactor, Map<V, Double> exogenousFactormap)
+    {
+        this(g, dampingFactor, exogenousFactormap, MAX_ITERATIONS_DEFAULT, TOLERANCE_DEFAULT);
+    }
+
+    /**
+     * Create and execute an instance of AlphaCentrality.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     * @param exogenousFactor the exogenous factor
+     * @param maxIterations the maximum number of iterations to perform
+     */
+    public AlphaCentrality(Graph<V, E> g, double dampingFactor, double exogenousFactor, int maxIterations)
+    {
+        this(g, dampingFactor, exogenousFactor, maxIterations, TOLERANCE_DEFAULT);
+    }
+
+    /**
+     * Create and execute an instance of AlphaCentrality.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     * @param exogenousFactormap the exogenous factor map
+     * @param maxIterations the maximum number of iterations to perform
+     */
+    public AlphaCentrality(Graph<V, E> g, double dampingFactor, Map<V, Double> exogenousFactormap, int maxIterations)
+    {
+        this(g, dampingFactor, exogenousFactormap, maxIterations, TOLERANCE_DEFAULT);
+    }
+
+    /**
+     * Create and execute an instance of AlphaCentrality.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     * @param exogenousFactor the exogenous factor
+     * @param maxIterations the maximum number of iterations to perform
+     * @param tolerance the calculation will stop if the difference of AlphaCentrality values between
+     *        iterations change less than this value
+     */
+    public AlphaCentrality(Graph<V, E> g, double dampingFactor, double exogenousFactor, int maxIterations, double tolerance)
+    {
+        this.g = g;
+        this.scores = new HashMap<>();
+
+        if (maxIterations <= 0) {
```

Comment:
Don't repeat the code for parameter validation.

Reply:
Can I do the parameter evaluation in the run method instead of the constructors...this way I will be evaluating parameters only once...

Reply:
Better to share a validation method.

Reply:
@jsichi Are you saying that @tibrewalpratik17 should extract the following code into a helper method, to prevent duplication across the two `AlphaCentrality` constructors?

```java
if (maxIterations <= 0) {
  throw new IllegalArgumentException("Maximum iterations must be positive");
}

if (dampingFactor < 0.0 || dampingFactor > 1.0) {
  throw new IllegalArgumentException("Damping factor not valid");
}

if (tolerance <= 0.0) {
  throw new IllegalArgumentException("Tolerance not valid, must be positive");
}
```